test/model.test.js

```diff
@@ -132,69 +122,81 @@ describe('loopback:model generator', function() {
-        plural: 'pds',
-      });
-
-      modelGen.run(function() {
-        var modelConfig = readModelsJsonSync();
-        expect(modelConfig.Product.dataSource).to.eql(null);
-        var product = readProductJsonSync();
-        expect(product).to.have.property('base', 'Model');
-        done();
-      });
-    });
-
-    it('should set dataSource to 1st one if db does not exist', function(done) {
-      wsModels.DataSourceDefinition.create({
-        name: 'db1',
-        connector: 'memory',
-        facetName: 'server',
-      }, function(err) {
-        if (err) return done(err);
-        var modelGen = givenModelGenerator();
-        helpers.mockPrompt(modelGen, {
-          name: 'Review',
-          plural: 'Reviews',
-        });
-
-        modelGen.run(function() {
+    it('should set dataSource to null', function() {
+      return helpers.run(path.join(__dirname, '../model'))
+        .cd(SANDBOX)
+        .withPrompts({
+          name: 'Product',
+          plural: 'pds',
+        }).then(function() {
           var modelConfig = readModelsJsonSync();
-          expect(modelConfig.Review.dataSource).to.eql('db1');
-          done();
+          expect(modelConfig.Product.dataSource).to.eql(null);
+          var product = readProductJsonSync();
+          expect(product).to.have.property('base', 'Model');
         });
-      });
     });
 
-    it('should set dataSource to db if it exists', function(done) {
-      wsModels.DataSourceDefinition.create([{
-        name: 'db1',
-        connector: 'memory',
-        facetName: 'server',
-      }, {
-        name: 'db',
-        connector: 'memory',
-        facetName: 'server',
-      }], function(err) {
-        if (err) return done(err);
-        var modelGen = givenModelGenerator();
-        helpers.mockPrompt(modelGen, {
-          name: 'Review',
-          plural: 'Reviews',
+    it('should set dataSource to 1st one if db does not exist', function() {
+      return new Promise(function(resolve, reject) {
+        wsModels.DataSourceDefinition.create({
+          name: 'db1',
+          connector: 'memory',
+          facetName: 'server',
+        }, function(err) {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(returnResult());
+          };
         });
+      });
 
-        modelGen.run(function() {
-          var modelConfig = readModelsJsonSync();
-          expect(modelConfig.Review.dataSource).to.eql('db');
-          done();
+      function returnResult() {
+        return helpers.run(path.join(__dirname, '../model'))
+          .cd(SANDBOX)
+          .withPrompts({
+            name: 'Review',
+            plural: 'Reviews',
+          }).then(function() {
+            var modelConfig = readModelsJsonSync();
+            expect(modelConfig.Review.dataSource).to.eql('db1');
+          });
+      }
+    });
+
+    it('should set dataSource to db if it exists', function() {
+      return new Promise(function(resolve, reject) {
```

Comment:
I think we can just return `wsModels.DataSourceDefinition.create`; I'm assuming the method has promise support

Reply:
Assuming loopback-workspace is depending on LB 3.x internally (and not on LB 2.x), then yes, methods like `create` should support promises out of the box.